src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -208,13 +204,35 @@ private void Commit(NodeCommitInfo nodeCommitInfo, bool skipSelf = false)
                 }
                 else
                 {
-                    List<NodeCommitInfo> nodesToCommit = new(16);
+                    Task CreateTaskForPath(TreePath childPath, TrieNode childNode, int idx)
+                    {
+                        return Task.Run(() =>
+                        {
+                            Commit(committer, ref childPath, new NodeCommitInfo(childNode!, node, idx), maxLevelForConcurrentCommit);
+                            committer.ReturnConcurrencyQuota();
+                        });
+                    }
+
+                    ArrayPoolList<Task>? childTasks = null;
+
```

Comment:
And here just have an `if` before and then do either for or Parallel.For

Reply:
It is slightly faster to call directly without creating intermediate TreePath to be put in a list. Thats why its done this way. Althought its probably does not matter.